### PR TITLE
Has been implemented fasthttp handler

### PR DESCRIPTION
**Pull request by akdilsiz**

### CODE REVIEW

**Comment by novln** [comment]
Hello, thank you for this pull request :slightly_smiling_face: 

I'll prefer that you provide an example such as these pull requests for **phi**: 
* #59 
* #41

I've moved the repository here if you want to submit a pull request: 
https://github.com/ulule/limiter-examples

With that being said, I'll gladly accept a pull request to add a new driver for **fasthttp** since it's low level and quite popular, if it's useful.

Cheers,

**Comment by akdilsiz** [comment]
@novln I sent PR for limiter-examples. https://github.com/ulule/limiter-examples/pull/1

**Comment by novln** [comment]
Sorry, that's not what I meant.

There is a lot of web framework that requires custom middleware, like **echo** or **beego** _(for example)_.
Unfortunately I cannot add them all because it will produce a bloated library.
_One could argue that I should remove Gin, and they may be right._

So I won't merge this pull request to add a **phi** middleware driver.

Nonetheless, you could create this middleware on your user profile and use it with your examples, if you like.

Cheers,

**Comment by akdilsiz** [comment]
@novln  I get it now. :) I can write a layer directly for fasthttp.

File: go.sum [hunk 1]
```diff
@@ -2,50 +2,44 @@ github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
+github.com/fate-lovely/phi v0.0.0-20171026043140-ee6510b82038 h1:N08k/Ufn2cMjpw1+hKDWNCcuiD9gQxRqvzl3Xi8SkLU=
```
Comment (novln) [inline go.sum]: Could you execute `go mod tiny` ?

File: drivers/middleware/fasthttp/middleware.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package fasthttp
+
+import (
+	"github.com/ulule/limiter/v3"
+	libFastHttp "github.com/valyala/fasthttp"
```
Comment (novln) [inline drivers/middleware/fasthttp/middleware.go]: The import alias should be lowercase.

File: drivers/middleware/fasthttp/middleware.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	for _, option := range options {
+		option.apply(middleware)
+	}
+
+	return middleware
+}
+
+// Handle gin request.
```
Comment (novln) [inline drivers/middleware/fasthttp/middleware.go]:
```suggestion
// Handle fasthttp request.
```

File: drivers/middleware/fasthttp/middleware_test.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
+package fasthttp_test
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/ulule/limiter/v3"
+	"github.com/ulule/limiter/v3/drivers/middleware/fasthttp"
+	"github.com/ulule/limiter/v3/drivers/store/memory"
```
Comment (novln) [inline drivers/middleware/fasthttp/middleware_test.go]: Could you re-organize these import ?

**Comment by novln** [comment]
Thank you for this contribution.

**Pull request merged**